addons/im_livechat/models/im_livechat_channel.py

```diff
@@ -298,26 +293,77 @@ def _get_livechat_discuss_channel_vals(self, chatbot_script=None, agent=None):
             'livechat_channel_id': self.id,
-            "livechat_failure": "no_answer" if agent else "no_failure",
+            "livechat_failure": "no_answer" if is_agent else "no_failure",
             "livechat_status": "in_progress",
-            'chatbot_current_step_id': chatbot_script._get_welcome_steps()[-1].id if chatbot_script else False,
+            'chatbot_current_step_id': chatbot_script._get_welcome_steps()[-1].id if is_chatbot_script else False,
             'channel_type': 'livechat',
-            'name': name,
+            'name': channel_name,
+        }
+
+    def _get_agent_member_vals(self, /, *, last_interest_dt, now, chatbot_script, operator_partner, operator_model, **kwargs):
+        return {
+            "chatbot_script_id": chatbot_script.id if operator_model == 'chatbot.script' else False,
+            "last_interest_dt": last_interest_dt,
+            "livechat_member_type": "agent" if operator_model == 'res.users' else "bot",
+            "partner_id": operator_partner.id,
+            "unpin_dt": now,
         }
 
+    def _get_channel_name(self, /, *, visitor_user=None, guest=None, agent, chatbot_script, operator_model, **kwargs):
+        if operator_model == 'chatbot.script':
+            channel_name = chatbot_script.title
+        else:
+            channel_name = ' '.join([
+                visitor_user.display_name if visitor_user else guest.name,
+                agent.livechat_username or agent.name
+            ])
+        return channel_name
+
+    def _get_operator_info(self, /, *, lang, country_id, previous_operator_id=None, chatbot_script_id=None, **kwargs):
+        agent = self.env['res.users']
+        chatbot_script = self.env['chatbot.script']
+        operator_partner = self.env['res.partner']
+        # The operator_model establishes the priority among potential operators (e.g., chatbot_script or ai_agent) for a live chat channel.
+        # It dictates which operator model is selected when multiple are configured.
+        operator_model = ''
+
+        if chatbot_script_id and chatbot_script_id in self.rule_ids.chatbot_script_id.ids:
+            chatbot_script = (
+                self.env["chatbot.script"]
+                .sudo()
+                .with_context(lang=self.env["chatbot.script"]._get_chatbot_language())
+                .search([("id", "=", chatbot_script_id)])
+            )
+            operator_partner = chatbot_script.operator_partner_id
+            operator_model = 'chatbot.script'
+
+        if not operator_model:
+            agent = self._get_operator(
+                previous_operator_id=previous_operator_id,
+                lang=lang,
+                country_id=country_id,
+            )
+            operator_partner = agent.partner_id
+            operator_model = 'res.users'
+
+        return {'agent': agent, 'chatbot_script': chatbot_script, 'operator_partner': operator_partner, 'operator_model': operator_model}
```

Comment:
returning operator_partner is useless, agent or chatbot_script is sufficient

Reply:
operator_model is useless too, recordset have models associated to them + as we have 2 different variables we can just check which one is returned